src/common/mfu_param_path.h

```diff
@@ -58,10 +58,10 @@ typedef struct {
     int                  fd;
 #ifdef DAOS_SUPPORT
     /* DAOS specific variables for I/O */
-    daos_off_t           offset;
-    dfs_obj_t*           obj;
-    dfs_t*               dfs;
-    struct d_hash_table* dfs_hash;
+    daos_off_t           offset;  /* file offset */
+    dfs_obj_t*           obj;     /* open object handle */
+    dfs_sys_t*           dfs_sys; /* handle for high-level file operations */
+    dfs_t*               dfs;     /* handle for lower-level file operations */
```

Comment:
@daltonbohning curious how dfs_t is still used?

Reply:
@dsikich It's used for any functions that either don't have a dfs_sys implementation, or for functions that a dfs_sys implementation would just be a direct wrapper for dfs. Right now, it's just truncate:
https://github.com/hpc/mpifileutils/blob/2a8e5c413c0aa15657cffc56b5173e8fef9d724f/src/common/mfu_io.c#L1120
But more generally, the idea is that we have full access to DFS, so we aren't restricted to the dfs_sys API. I.e. instead of _**replacing**_ the DFS API, the DFS Sys API is built _**on top of**_ DFS